src/color.cpp

```diff
@@ -6,756 +6,412 @@
-   case 65: return c_dkgray_white;
-   case 66: return c_ltgray_white;
-   case 67: return c_white_white;
-   case 68: return c_red_white;
-   case 69: return c_ltred_white;
-   case 70: return c_green_white;
-   case 71: return c_ltgreen_white;
-   case 72: return c_brown_white;
-   case 73: return c_yellow_white;
-   case 74: return c_blue_white;
-   case 75: return c_ltblue_white;
-   case 76: return c_magenta_white;
-   case 77: return c_pink_white;
-   case 78: return c_cyan_white;
-   case 79: return c_ltcyan_white;
-
-   case 80: return c_black_green;
-   case 81: return c_dkgray_green;
-   case 82: return c_ltgray_green;
-   case 83: return c_white_green;
-   case 84: return c_red_green;
-   case 85: return c_ltred_green;
-   case 86: return c_green_green;
-   case 87: return c_ltgreen_green;
-   case 88: return c_brown_green;
-   case 89: return c_yellow_green;
-   case 90: return c_blue_green;
-   case 91: return c_ltblue_green;
-   case 92: return c_magenta_green;
-   case 93: return c_pink_green;
-   case 94: return c_cyan_green;
-   case 95: return c_ltcyan_green;
-
-   case 96: return c_black_yellow;
-   case 97: return c_dkgray_yellow;
-   case 98: return c_ltgray_yellow;
-   case 99: return c_white_yellow;
-   case 100: return c_red_yellow;
-   case 101: return c_ltred_yellow;
-   case 102: return c_green_yellow;
-   case 103: return c_ltgreen_yellow;
-   case 104: return c_brown_yellow;
-   case 105: return c_yellow_yellow;
-   case 106: return c_blue_yellow;
-   case 107: return c_ltblue_yellow;
-   case 108: return c_magenta_yellow;
-   case 109: return c_pink_yellow;
-   case 110: return c_cyan_yellow;
-   case 111: return c_ltcyan_yellow;
-
-   case 112: return c_black_magenta;
-   case 113: return c_dkgray_magenta;
-   case 114: return c_ltgray_magenta;
-   case 115: return c_white_magenta;
-   case 116: return c_red_magenta;
-   case 117: return c_ltred_magenta;
-   case 118: return c_green_magenta;
-   case 119: return c_ltgreen_magenta;
-   case 120: return c_brown_magenta;
-   case 121: return c_yellow_magenta;
-   case 122: return c_blue_magenta;
-   case 123: return c_ltblue_magenta;
-   case 124: return c_magenta_magenta;
-   case 125: return c_pink_magenta;
-   case 126: return c_cyan_magenta;
-   case 127: return c_ltcyan_magenta;
-
-   case 128: return c_black_cyan;
-   case 129: return c_dkgray_cyan;
-   case 130: return c_ltgray_cyan;
-   case 131: return c_white_cyan;
-   case 132: return c_red_cyan;
-   case 133: return c_ltred_cyan;
-   case 134: return c_green_cyan;
-   case 135: return c_ltgreen_cyan;
-   case 136: return c_brown_cyan;
-   case 137: return c_yellow_cyan;
-   case 138: return c_blue_cyan;
-   case 139: return c_ltblue_cyan;
-   case 140: return c_magenta_cyan;
-   case 141: return c_pink_cyan;
-   case 142: return c_cyan_cyan;
-   case 143: return c_ltcyan_cyan;
-
- }
- return c_black;
+    const std::string sName = all_colors.get(color);
+
+    if ( mapInvert[sName].first == "" ) {
```

Comment:
Btw. `std::string` has a function `empty()` which does this test, is shorter to read and write and probably a tiny bit faster.